src/c_api.cpp

```diff
@@ -1674,6 +1754,9 @@ int LGBM_BoosterPredictForMats(BoosterHandle handle,
   if (config.num_threads > 0) {
     omp_set_num_threads(config.num_threads);
   }
+
+  AdditionalConfig(&config);
+
```

Comment:
can we move all these `AdditionalConfig` into config.cpp? Like in  https://github.com/microsoft/LightGBM/blob/2792923d35e7f3d392474ceffe2185c661047b2e/src/io/config.cpp#L191-L243